src/installer.ts

```diff
@@ -1,193 +1,216 @@
-    } else {
-      throw new Error('Download URL is neither a zip file nor a tarball');
-    }
+    static async downloadAndExtract(url: string, versionToCache: BinaryVersion | undefined): Promise<FindBinaryStatus> {
+        let downloadDir = "";
+        const pathOcArchive = await tc.downloadTool(url);
+        if (url.endsWith("zip")) {
+            downloadDir = await tc.extractZip(pathOcArchive);
+        }
+        else if (url.endsWith("tar") || url.endsWith("tar.gz")) {
+            downloadDir = await tc.extractTar(pathOcArchive);
+        }
+        else {
+            throw new Error("Download URL is neither a zip file nor a tarball");
+        }
 
-    const ocBinaryFilename: string = utils.getOcBinaryFilename();
-    const ocBinaryPath = await Installer.findOcFile(downloadDir, ocBinaryFilename);
-    if (!ocBinaryPath) {
-      return {
-        found: false,
-        reason: `An error occurred while downloading and extracting oc binary from ${url}. File ${ocBinaryFilename} not found.`
-      };
-    }
-    await promisify(fs.chmod)(ocBinaryPath, '0755');
-    if (versionToCache) {
-      await tc.cacheFile(ocBinaryPath, path.parse(ocBinaryPath).base, 'oc', versionToCache.version);
-      core.info(`Added oc ${versionToCache.version} to tool cache`);
-    }
-    core.info(`oc installed to ${ocBinaryPath}`);
-    return {
-      found: true,
-      path: ocBinaryPath
-    };
-  }
-
-  static getOcURLToDownload(version: BinaryVersion): string {
-    const major = version.major;
-    if (major !== 3 && major !== 4) {
-      throw new Error('Major version must be "3" or "4". No other versions of oc are supported.');
+        const ocBinaryFilename: string = utils.getOcBinaryFilename();
+        const ocBinaryPath = await Installer.findOcFile(downloadDir, ocBinaryFilename);
+        if (!ocBinaryPath) {
+            return {
+                found: false,
+                reason: `An error occurred while downloading and extracting oc binary from ${url}. `
+                + `File ${ocBinaryFilename} not found.`,
+            };
+        }
+        await promisify(fs.chmod)(ocBinaryPath, "0755");
+        if (versionToCache) {
+            await tc.cacheFile(ocBinaryPath, path.parse(ocBinaryPath).base, "oc", versionToCache.version);
+            core.info(`Added oc ${versionToCache.version} to tool cache`);
+        }
+        core.info(`oc installed to ${ocBinaryPath}`);
+        return {
+            found: true,
+            path: ocBinaryPath,
+        };
     }
 
-    const url = `${utils.getDownloadSite(major)}/${version.version}/${utils.getZipPath()}`;
-    core.info(`Found oc version ${version.version} at ${url}`);
-    return url;
-  }
+    static getOcURLToDownload(version: BinaryVersion): string {
+        const major = version.major;
+        if (major !== 3 && major !== 4) {
+            throw new Error("Major version must be \"3\" or \"4\". No other versions of oc are supported.");
```

Comment:
instead of escaping the double quotes, i would have just used backticks for the whole string